File: plugins/methods/src/com/intellij/methods/vfs/MethodsFileRegistry.java
package com.intellij.methods.vfs;

import org.jetbrains.annotations.NotNull;

import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.ConcurrentMap;

/**
 * In-memory registry mapping path strings to {@link MethodsVirtualFile} instances.
 */
public final class MethodsFileRegistry {
  private static final ConcurrentMap<String, MethodsVirtualFile> FILES = new ConcurrentHashMap<>();

  private MethodsFileRegistry() {}

  public static @NotNull MethodsVirtualFile getOrCreate(@NotNull String path) {
    return FILES.computeIfAbsent(path, MethodsVirtualFile::new);
  }

  public static void update(@NotNull String path, byte @NotNull [] bytes, long stamp) {
    getOrCreate(path).updateContent(bytes, stamp);
  }
}

File: plugins/methods/src/com/intellij/methods/vfs/MethodsFileRegistry.kt
package com.intellij.methods.vfs

import java.util.concurrent.ConcurrentHashMap

/**
 * In-memory registry mapping path strings to [MethodsVirtualFile] instances.
 * This class can later be moved into a standalone plugin.
 */
internal object MethodsFileRegistry {
  private val files = ConcurrentHashMap<String, MethodsVirtualFile>()

  fun getOrCreate(path: String): MethodsVirtualFile =
    files.computeIfAbsent(path) { MethodsVirtualFile(it) }

  fun update(path: String, bytes: ByteArray, stamp: Long) {
    getOrCreate(path).updateContent(bytes, stamp)
  }
}
